plugins/woocommerce/src/Admin/API/Reports/DataStore.php

```diff
@@ -160,6 +160,54 @@ public function __construct() {
 		}
 	}
 
+
+	/**
+	 * Get the data based on args.
+	 *
+	 * Returns the report data based on parameters supplied by the user.
+	 * Fetches it from cache or returns `get_noncached_data` result.
+	 *
+	 * @param array $query_args Query parameters.
+	 * @return stdClass|WP_Error
+	 */
+	public function get_data( $query_args ) {
+		$defaults   = $this->get_default_query_vars();
+		$query_args = wp_parse_args( $query_args, $defaults );
+		$this->normalize_timezones( $query_args, $defaults );
+
+		/*
+		 * We need to get the cache key here because
+		 * parent::update_intervals_sql_params() modifies $query_args.
+		 */
+		$cache_key = $this->get_cache_key( $query_args );
+		$data      = $this->get_cached_data( $cache_key );
+
+		if ( false === $data ) {
+			$data = $this->get_noncached_data( $query_args );
+			$this->set_cached_data( $cache_key, $data );
+		}
+
+		return $data;
+	}
+
+	/**
+	 * Get the default query arguments to be used by get_data().
+	 * These defaults are only partially applied when used via REST API, as that has its own defaults.
+	 *
+	 * @return array Query parameters.
+	 */
+	public function get_default_query_vars() {
+		return array(
+			'per_page' => get_option( 'posts_per_page' ),
+			'page'     => 1,
+			'order'    => 'DESC',
+			'orderby'  => 'date',
+			'before'   => TimeInterval::default_before(),
+			'after'    => TimeInterval::default_after(),
+			'fields'   => '*',
+		);
+	}
```

Comment:
:scroll: I wonder if we really need to handle default query vars here and in the Query class. Maybe we could simplify it to just one source.